pkg/controller/podautoscaler/horizontal.go

```diff
@@ -229,13 +229,27 @@ func (a *HorizontalController) processNextWorkItem() bool {
 	return true
 }
 
+// Types to help keep status and spec separate and avoid a positive
+// feedback loop.  Generally, nothing from status should go back into
+// spec.
+type specReplicas int32
+type statusReplicas int32
+
+// Many functions shadow the specReplicas type with a specReplicas
+// argument to prevent unintentional crosstalk between spec and status
+// replica counts.  This function allows casting from int32 to
+// specReplicas when generating scale recommendations.
+func newSpecReplicas(i int32) specReplicas {
+	return specReplicas(i)
+}
+
 // computeReplicasForMetrics computes the desired number of replicas for the metric specifications listed in the HPA,
 // returning the maximum of the computed replica counts, a description of the associated metric, and the statuses of
 // all metrics computed.
 func (a *HorizontalController) computeReplicasForMetrics(hpa *autoscalingv2.HorizontalPodAutoscaler, scale *autoscalingv1.Scale,
-	metricSpecs []autoscalingv2.MetricSpec) (replicas int32, metric string, statuses []autoscalingv2.MetricStatus, timestamp time.Time, err error) {
+	metricSpecs []autoscalingv2.MetricSpec) (replicas specReplicas, metric string, statuses []autoscalingv2.MetricStatus, timestamp time.Time, err error) {
 
-	currentReplicas := scale.Status.Replicas
+	specReplicas := specReplicas(scale.Spec.Replicas)
```

Comment:
What implication does this change has?

Reply:
Nevermind, it's explained in the PR description :).

Reply:
This is the root of the bug.  Writing status back into spec.